packages/maskbook/src/extension/injected-script/etherumChange.ts

```diff
@@ -0,0 +1,30 @@
+import type { ChainId } from '@masknet/web3-shared'
+import type Web3 from 'web3'
+
+type EthereumInjectedContext = {
+    ethereum:
+        | (Web3 & {
+              isMetaMask: boolean
+              on: (eventName: string, callback: (...args: unknown[]) => void) => void
+          })
+        | null
+}
+
+function registerListenersOnMetaMaskProvider() {
+    const ethereum = (globalThis as unknown as EthereumInjectedContext).ethereum
+    if (!ethereum?.isMetaMask) return
+    ethereum.on('accountsChanged', onMetaMaskAccountsChange as (...args: unknown[]) => void)
+    ethereum.on('chainChanged', onMetaMaskChainIdChange as (...args: unknown[]) => void)
+}
+
+function onMetaMaskAccountsChange(accounts: string) {
+    console.log(`Accounts: ${accounts}`)
+    window.open('chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/index.html#/', '_blank', 'noopener noreferrer')
```

Comment:
🤔 can you explain this?

Reply:
The users thought Mask Network connected theirs accounts with Twitter. So they switch accounts on the Twitter page. This handler would open the dashboard page if it detected the user switched their accounts in the wrong place. A dialog explains the right way of accounts management with MetaMask will be better.

Reply:
But I don't see a URL like `index.html#/explain-metamask`. It seems like it jumps to the dashboard directly.

Reply:
Actually, this URL is not allowed to open in this way. So I think an injected dialog will be better.